### PR TITLE
Add Remove Accidental Hits layer.

### PR DESCRIPTION
**Pull request by hertschuh**
This layer zeroes the logits of accidental negatives.

### CODE REVIEW

File: keras_rs/src/layers/retrieval/remove_accidental_hits.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    ) -> types.Tensor:
+        """Zeroes selected logits.
+
+        For each row in the batch, zeroes the logits of negative candidates that
+        have the same ID as the positive candidate in that row.
+
+        Args:
+          labels: one-hot labels tensor, typically
+              `[batch_size, num_candidates]` but can have more dimensions or be
+              1d as `[num_candidates]`.
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/remove_accidental_hits.py:34]: 1d --> 1D

File: keras_rs/src/layers/retrieval/remove_accidental_hits.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        labels_shape = ops.shape(labels)
+        labels_rank = len(labels_shape)
+        logits_shape = ops.shape(logits)
+        candidate_ids_shape = ops.shape(candidate_ids)
+        candidate_ids_rank = len(candidate_ids_shape)
+
+        if not keras_utils.check_shapes_compatible(labels_shape, logits_shape):
+            raise ValueError(
+                "The shape of `labels` and `logits` must match. Got "
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/remove_accidental_hits.py:56]: Nit: let's follow the same format everywhere: https://github.com/keras-team/keras-rs/blob/7553d191a1a25c39976b0494b439b66477a9c23d/keras_rs/src/utils/pairwise_loss_utils.py#L82. So, should be "Received: ..."

File: keras_rs/src/layers/retrieval/remove_accidental_hits.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        have the same ID as the positive candidate in that row.
+
+        Args:
+          labels: one-hot labels tensor, typically
+              `[batch_size, num_candidates]` but can have more dimensions or be
+              1d as `[num_candidates]`.
+          logits: logits tensor. Must have the same shape as `labels`.
+          candidate_ids: candidate identifiers tensor, can be `[num_candidates]`
+              or `[batch_size, num_candidates]` or have more dimensions as long
+              as they match the last dimensions of `labels`.
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/remove_accidental_hits.py:38]: Hmmm, the other layers (`SamplingProbability`, `HardNegativeMining`) don't have the option to have extra batch dimensions. Should we do this for uniformity?

**Review by abheesht17 — COMMENTED** [review]
Thanks for the PR!

File: keras_rs/src/layers/retrieval/remove_accidental_hits_test.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        return labels, logits, candidate_ids
+
+    @parameterized.named_parameters(
+        [
+            {"testcase_name": "1_1", "logits_rank": 1, "candidate_ids_rank": 1},
+            {"testcase_name": "2_1", "logits_rank": 2, "candidate_ids_rank": 1},
+            {"testcase_name": "2_2", "logits_rank": 2, "candidate_ids_rank": 2},
+            {"testcase_name": "3_1", "logits_rank": 3, "candidate_ids_rank": 1},
+            {"testcase_name": "3_2", "logits_rank": 3, "candidate_ids_rank": 2},
+            {"testcase_name": "3_3", "logits_rank": 3, "candidate_ids_rank": 3},
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/remove_accidental_hits_test.py:38]: Should we rename the UTs to `logits_rank_{x}_candidate_ids_rank_{y}`?

File: keras_rs/src/layers/retrieval/remove_accidental_hits.py (hunk 3 above)
Comment (hertschuh) [reply keras_rs/src/layers/retrieval/remove_accidental_hits.py:38]: Hmm... yeah, since they're used together.

I noticed it was easy to generalize so I just did it (although it made unit tests trickier).

I'll do it in a separate PR. https://github.com/keras-team/keras-rs/issues/39

**Review by abheesht17 — APPROVED** [review]
Thanks for the PR! Looks good to me. Sorry for the delay in reviewing.

Note: We need to add doc-string examples. I've created an issue, and will do it at some point for all APIs: https://github.com/keras-team/keras-rs/issues/33

File: keras_rs/src/layers/retrieval/remove_accidental_hits.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                "`candidate_ids` should have the same shape as the last "
+                "dimensions of `labels`. Received: "
+                f"`candidate_ids.shape` = {candidate_ids_shape}, "
+                f"`labels.shape` = {labels_shape}."
+            )
+
+        # Add dimensions to `candidate_ids` to have the same rank as `labels`.
+        if candidate_ids_rank < labels_rank:
+            candidate_ids = ops.expand_dims(
+                candidate_ids, list(range(labels_rank - candidate_ids_rank))
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras_rs/src/layers/retrieval/remove_accidental_hits.py:74]: Yes, `tf.expand_dims` doesn't support this, but the Numpy API does (so it's implemented with a loop on TF).
